lib/_tls_common.js

```diff
@@ -132,6 +134,52 @@ exports.createSecureContext = function createSecureContext(options, context) {
     }
   }
 
+  if (options.pskCallback && c.context.enablePskCallback) {
+    c.context.onpskexchange = (identity, maxPskLen, maxIdentLen) => {
+      let ret = options.pskCallback(identity);
+      if (typeof ret !== 'object') {
+        ret = undefined;
+      }
+
+      if (ret) {
+        ret = { psk: ret.psk, identity: ret.identity };
+
+        if (typeof ret.psk === 'string') {
+          ret.psk = Buffer.from(ret.psk);
+        } else if (!Buffer.isBuffer(ret.psk)) {
+          throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'psk',
+                                     ['string', 'buffer']);
+        }
+        if (ret.psk.length > maxPskLen) {
+          throw new errors.RangeError(
+            'ERR_ASSERTION',
+            `Pre-shared key exceed ${maxPskLen} bytes`);
+        }
+
+        // Only set for the client callback, which must provide an identity.
+        if (maxIdentLen) {
+          if (typeof ret.identity === 'string') {
+            ret.identity = Buffer.from(ret.identity + '\0');
```

Comment:
Well, the C++ method is using the identity in its buffer form, so we could pass a string and then convert it to a buffer in C++. Since `Buffer::Data` returns a `char*`, that's directly compatible with `identity`. I suppose we could convert the `string` directly to a `char*` as well. Seems like it would be about the same amount of work on either side.